socialdistribution/api/views.py

```diff
@@ -0,0 +1,113 @@
+from django.http import HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+
+from profiles.models import Author
+from posts.models import Post, Comment
+
+from .utils import post_to_dict, comment_to_dict
+
+
+# Create your views here.
+@csrf_exempt
+def all_posts(request):
+    # this view only accepts GET, 405 Method Not Allowed for other methods
+    if request.method != "GET":
+        return HttpResponse(f"Method {request.method} not allowed", status=405)
+
+    # get a list of all "PUBLIC" visibility posts on our node
+    public_posts = Post.objects.filter(visibility='PUBLIC')
+
+    # response body - to be converted into JSON and returned in the response
+    response_body = {
+        "query" : "posts",
+        "count" : public_posts.count(),
+        "size"  : "IMPLEMENT PAGINATION",
+        "next"  : "IMPLEMENT PAGINATION",
+        "previous" : "IMPLEMENT PAGINATION",
+        "posts" : [post_to_dict(post) for post in public_posts]
+    }
+
+    return JsonResponse(response_body)
+
+@csrf_exempt
+def single_post(request, post_id):
+    posts = Post.objects.filter(id=post_id)
+    
+    # GET a post which doesn't exist - 404 Not Found
+    if request.method == "GET" and posts.count() == 0:
+        return HttpResponse("That post does not exist", status=404)
+
+    # POST (insert) a post which already exists - 403 Forbidden
+    if request.method == "POST" and posts.count() > 0:
+        return HttpResponse("That post already exists", status=403)
+
+    # GET a post which exists - return post in JSON format
+    if request.method == "GET" and posts.count() > 0:
+        response_body = {
+            "query" : "posts",
+            "post" : post_to_dict(posts[0])
+        }
+
+        return JsonResponse(response_body)
+
+    # TODO
+    # POST / PUT a post which doesn't exist - insert post
+    if request.method in ("POST", "PUT") and posts.count() == 0:
+        return HttpResponse("Insert post")
+
+    # TODO
+    # PUT a post which exists - update post
+    if request.method == "PUT" and posts.count() > 0:
+        return HttpResponse("Update post")
+
+@csrf_exempt
+def author_posts(request, author_id):
+    # this view only accepts GET, 405 Method Not Allowed for other methods
+    if request.method != "GET":
+        return HttpResponse(f"Method {request.method} not allowed", status=405)
+
+    authors = Author.objects.filter(id=author_id)
+
+    # author does not exist - 404 Not Found
+    if authors.count() == 0:
+        return HttpResponse("That author does not exist", status=404)
+
```

Comment:
I wonder the text for HTTP would be "Page not found"??

Reply:
A 404 is technically a Not Found, and in this context, it makes sense to use a 404 to indicate that the author does not exist.

Reply:
Yeah you're looking for something which doesn't exist, makes sense to return a 404 Not Found. The response body is mainly for debugging purposes, since you won't ever be loading the api url as a webpage.